examples/c/ex_col_store.c

```diff
@@ -255,73 +255,37 @@ static int
 find_min_and_max_temp(
   WT_SESSION *session, uint16_t start_time, uint16_t end_time, int *min_temp, int *max_temp)
 {
-    WT_CURSOR *end_time_cursor, *join_cursor, *start_time_cursor;
+    WT_CURSOR *cursor;
     WT_DECL_RET;
     uint64_t recno;
-    int exact;
     uint16_t hour;
     uint8_t temp;
 
-    /*! [col-store join] */
-
-    /* Open cursors needed by the join. */
-    error_check(
-      session->open_cursor(session, "join:table:weather(hour,temp)", NULL, NULL, &join_cursor));
-    error_check(
-      session->open_cursor(session, "index:weather:hour", NULL, NULL, &start_time_cursor));
-    error_check(session->open_cursor(session, "index:weather:hour", NULL, NULL, &end_time_cursor));
-
-    /*
-     * Select values WHERE (hour >= start AND hour <= end). Find the starting record closest to
-     * desired start time.
-     */
-    start_time_cursor->set_key(start_time_cursor, start_time);
-    error_check(start_time_cursor->search_near(start_time_cursor, &exact));
-    if (exact == -1) {
-        ret = start_time_cursor->next(start_time_cursor);
-        if (ret == WT_NOTFOUND)
-            return ret;
-        else
-            error_check(ret);
-    }
-
-    error_check(session->join(session, join_cursor, start_time_cursor, "compare=ge"));
-
-    /* Find the ending record closest to desired end time. */
-    end_time_cursor->set_key(end_time_cursor, end_time);
-    error_check(end_time_cursor->search_near(end_time_cursor, &exact));
-    if (exact == 1) {
-        ret = end_time_cursor->prev(end_time_cursor);
-        if (ret == WT_NOTFOUND)
-            return ret;
-        else
-            error_check(ret);
-    }
-
-    error_check(session->join(session, join_cursor, end_time_cursor, "compare=le"));
+    /* Open the cursor. */
+    error_check(session->open_cursor(session, "table:weather(hour,temp)", NULL, NULL, &cursor));
 
     /* Initialize minimum temperature and maximum temperature to temperature of the first record. */
-    ret = join_cursor->next(join_cursor);
+    ret = cursor->next(cursor);
     if (ret == WT_NOTFOUND)
         return ret;
     else
         error_check(ret);
 
-    error_check(join_cursor->get_key(join_cursor, &recno));
-    error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+    error_check(cursor->get_key(cursor, &recno));
+    error_check(cursor->get_value(cursor, &hour, &temp));
     *min_temp = temp;
     *max_temp = temp;
 
-    /* Iterating through found records between start and end time to find the min & max temps. */
```

Comment:
Since we are changing from iterating through found records to iterating through all records, would there be a chance we hit `WT_NOTFOUND` here? And if no records are found, can we still succeed since we won't be returning `ret = WT_NOTFOUND` after this point?

I think the function description - "If no records are found it will return WT_NOTFOUND" , needs to be updated as well.

Reply:
No. The no records at all case is handled on lines 304-306 when it is initializing *min_temp and *max_temp from the first key/value found. The function description does not have to change since WT_NOFOUND is returned on line 306.